### PR TITLE
Replaced the habit create function with update_or_create method in (h…

### PR DESCRIPTION
**Pull request by hienhihi**
…abit_create) in views.py. Also replaced log_entry_create with logged variable. No longer have bug when create new habit and add note

### CODE REVIEW

File: webapp/habits/views.py [hunk 1]
```diff
@@ -38,26 +38,24 @@ def habit_create(request):
...
+            id=request.POST.get('habit_id'),
+            defaults={
+                'user':request.user,
+                'name': request.POST.get('name'),
+                'description': request.POST.get('description'),
+                'schedule':request.POST.get('schedule'),
+                'color':request.POST.get('color'),
+                }
+                )
+    messages.success(request, 'Habit added!')
```
Comment (stephencockerill) [inline webapp/habits/views.py]: this part looks good :+1:

File: webapp/habits/views.py [hunk 2]
```diff
@@ -38,26 +38,24 @@ def habit_create(request):
...
 
 def log_entry_create(request, habit_id):
     if request.POST:
         habit = Habit.objects.get(id=habit_id)
-        LogEntry.objects.create(
-            note=request.POST.get('note'),
-            logged=request.POST.get('logged'),
-            habit=habit,
-        )
+        logged=request.POST.get('logged', datetime.now())
```
Comment (stephencockerill) [inline webapp/habits/views.py]: We still want to create a LogEntry, but if there is no `logged` value in `request.POST` we want to default to the current time. So you want to keep the `LogEntry.objects.create()` call, but updae the `logged=request.POST.get('logged')` line to what you have here.

File: webapp/models.py [hunk 3]
```diff
@@ -44,4 +44,6 @@ class LogEntry(models.Model):
     habit = models.ForeignKey(
         Habit,
         on_delete = models.CASCADE,
+        null=False,
+        blank=False
```
Comment (stephencockerill) [inline webapp/models.py]: oh and since you are making a change to the model, make sure to makemigrations and add that to this commit

Comment (hienhihi) [reply webapp/models.py]: yup, did that

Comment (stephencockerill) [reply webapp/models.py]: hmm I think you did it locally but maybe didn't add the new migration file to this commit cause I'm not seeing it in the files changed tab

Comment (hienhihi) [reply webapp/models.py]: I did git push after changing the codes. Let me figure out how to get it up here

Comment (stephencockerill) [reply webapp/models.py]: should be another file in this folder: https://github.com/habitfarm/habitfarm/tree/master/webapp/migrations

Comment (stephencockerill) [reply webapp/models.py]: if you do `git status` on this branch, do you see one untracked file for the migration? If so, you can `git add` the file, then commit, then push and it should pop up here. Also, did you mean to close this PR?

**Pull request closed without merging**